src/commandSocket.cpp

```diff
@@ -91,11 +91,59 @@ class CommandSocket::Private
 
     std::unordered_map<int, std::shared_ptr<cura::proto::Layer>> sliced_layers;
 };
+
+class CommandSocket::PolygonCompiler
```

Comment:
Could you please document this class and all its members?

It's unclear to me what the code is for.